max2max/internal/loader/append.go

```diff
@@ -1,25 +1,48 @@
 package loader
 
 import (
+	"context"
 	"fmt"
 	"log/slog"
 	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
 )
 
 type appendLoader struct {
-	logger *slog.Logger
+	logger       *slog.Logger
+	meter        metric.Meter
+	queryCounter metric.Int64Counter
 }
 
-func NewAppendLoader(logger *slog.Logger) *appendLoader {
-	return &appendLoader{
-		logger: logger,
+func NewAppendLoader(logger *slog.Logger) (*appendLoader, error) {
+	meter := otel.Meter("loader")
+	queryCounter, err := meter.Int64Counter("query.count")
+	if err != nil {
+		return nil, err
 	}
+
+	return &appendLoader{
+		logger:       logger,
+		meter:        meter,
+		queryCounter: queryCounter,
+	}, nil
 }
 
 func (l *appendLoader) GetQuery(tableID, query string) string {
+	l.queryCounter.Add(context.Background(), 1, metric.WithAttributes(
+		attribute.String("method", "GetQuery"),
+		attribute.String("tableID", tableID),
+	))
```

Comment:
Do we need a telemetry for this function ? I does not include any external calls.

Reply:
this is just a sample for sending telemetry metrics to collector, I'll remove it as it's not necessary